### ISSUE TITLE
About the PCS

### ISSUE THREAD

**Issue by GuoQingqing** [weijun88/SANet#1]
Hi,
you have done a splendid work  for Polyp Segmentation, and  I have a question about PCS.  In  section 3.3, you proposed a Probability Correction Strategy  to explicitly correct the predicted probability through logit reweighting. 
However,  the logit of positive samples(i.e., logit > 0) is still greater than 0 after the normalization, and the result after sigmoid is greater than 0.5, so it is still recognized as a positive sample.  I cannot understand your intention to use the PCS. 

It would be very kind of you to answer my question.
Thanks.

**Comment by weijun-arc** [comment]
Hello,  PCS is not designed for transforming postive sample to negtive sample or vice versa. PCS devotes to enhance the prediction from low confidence to high confidence. For example, a positve sample , logit equal to 0.01, corresponds to 0.5 after sigmoid. However, PCS could make the corresponding value equal to 1. So the boundary will be clearer. That's goal of PCS.

**Comment by GuoQingqing** [comment]
Thank you very much! I understand. The final prediction in  in the code is generated using **np.round(pred)**, so I thought the final prediction was converted to the label '0 and1'.  May I ask  wheather the calculation of mDice and mIoU  adopts the predicted probability  or the predicted label ?

**Comment by weijun-arc** [comment]
The evaluation codes are borrowed from the PraNet for a fair comparison. I think if the authors want to transform the predicted probability into the binary mask, they need to determine the threshold first and this will bring a hyper-parameter. Though 0 may be the a good threshold, but they could not be sure until they did the experiment. So directly using the predcited probability is the easiest way.

**Comment by GuoQingqing** [comment]
> The evaluation codes are borrowed from the PraNet for a fair comparison. I think if the authors want to transform the predicted probability into the binary mask, they need to determine the threshold first and this will bring a hyper-parameter. Though 0 may be the a good threshold, but they could not be sure until they did the experiment. So directly using the predcited probability is the easiest way.

I got it. Thanks a lot!

**Comment by nguyenlecong** [comment]
> Thank you very much! I understand. The final prediction in in the code is generated using **np.round(pred)**, so I thought the final prediction was converted to the label '0 and1'. May I ask wheather the calculation of mDice and mIoU adopts the predicted probability or the predicted label ?

The final prediction is generated using np.round(pred) **after multiplied by 255.** So it is still **grayscale**.
